frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/SelectionManager.java

```diff
@@ -192,8 +192,8 @@ public synchronized SelectionResult selectSingleNode(String taskRoleName) throws
   }
 
   public synchronized SelectionResult select(String taskRoleName)
-      throws NotAvailableException {
-    ResourceDescriptor requestResource = requestManager.getTaskResources().get(taskRoleName);
+      throws Exception {
```

Comment:
>Exception [](start = 13, length = 9)

Why change exception type from the more specific NotAvaliableException to such as general exceptoin?